app/models/assignment.rb

```diff
@@ -777,8 +777,11 @@ def summary_csv(role)
     headers[1] << ''
 
     result_ids = groupings.pluck('results.id').uniq.compact
-    total_marks_hash = Result.get_total_marks(result_ids)
-    extra_marks_hash = Result.get_total_extra_marks(result_ids, max_mark: max_mark)
+    subtotals = Result.get_subtotals(result_ids)
+    extra_marks_hash = Result.get_total_extra_marks(result_ids, max_mark: max_mark, subtotals: subtotals)
+    total_marks_hash = subtotals.merge(extra_marks_hash) do |_result_id, subtotal, extra|
```

Comment:
Overall we should try to avoid duplicating logic where possible. In a similar vein as what I suggested earlier, please modify the `Result.get_total_marks` method to optionally take in both `subtotals` and `extra_marks` hashes (default `nil`), and reuse those to compute the result.

This will allow you to keep the `total_marks_hash = Result.get_total_marks(result_ids)` call, just with extra arguments.